File: src/war3map/Generic/DataContainers/DelayDto.lua
local ____exports = {}
____exports.DelayDto = {}
local DelayDto = ____exports.DelayDto
DelayDto.name = "DelayDto"
DelayDto.__index = DelayDto
DelayDto.prototype = {}
DelayDto.prototype.__index = DelayDto.prototype
DelayDto.prototype.constructor = DelayDto
function DelayDto.new(...)
    local self = setmetatable({}, DelayDto.prototype)
    self:____constructor(...)
    return self
end
function DelayDto.prototype.____constructor(self, func, delay)
    self.age = 0
    self["function"] = func
    self.delay = delay
end
return ____exports

File: TreeLib/Utility/DelayDto.ts
export class DelayDto {
    public function: Function;
    public delay: number;
    public age: number = 0;
    public repeats: number;
    public repeatCounter: number;
    constructor(func: Function, delay: number, repeats: number) {
        this.function = func;
        this.delay = delay;
        this.repeats = repeats;
        this.repeatCounter = 0;
    }
}